software/contrib/euclidean_rhythm.py

```diff
@@ -0,0 +1,356 @@
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+12, 2, box_size[1]-12, 1)
+                    elif len(self.notes_list) < 20:
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+12, 1, box_size[1]-12, 1)
+            else:
+                # High
+                if self.notes_list[i] == 1:
+                    if len(self.notes_list) < 10:
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+8, 3, box_size[1]-8, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+5, 3, 2, 1)
+                    elif len(self.notes_list) < 14:
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+8, 2, box_size[1]-8, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+5, 2, 2, 1)
+                    elif len(self.notes_list) < 20:
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+8, 1, box_size[1]-8, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+5, 1, 2, 1)
+                # Low
+                else:
+                    if len(self.notes_list) < 10:
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+15, 3, box_size[1]-15, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+12, 3, 2, 1)
+                    elif len(self.notes_list) < 14:
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+15, 2, box_size[1]-15, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+12, 2, 2, 1)
+                    elif len(self.notes_list) < 20:
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+15, 1, box_size[1]-15, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+12, 1, 2, 1)
+
+class EuclideanRhythm(EuroPiScript):
+    def __init__(self):
+        # length, fill, offset, cv
+        self.notes_para = [[1, 0, 0, 0], [1, 0, 0, 0], [1, 0, 0, 0],
+                           [1, 0, 0, 0], [1, 0, 0, 0], [1, 0, 0, 0]]
+        
+        # length, fill, offset
+        self.notes_para_final = [[1, 0, 0], [1, 0, 0], [1, 0, 0],
+                                 [1, 0, 0], [1, 0, 0], [1, 0, 0]]
+
+        
+        self.selection_idx = 0 # change editing gates
+        self.editing_idx = 0 # change editing parameters
+        self.blink_flg = 0 # blink selection with tempo
+        self.clock_flg = 0 # 0: internal clock, 1: external clock (from digital in)
```

Comment:
If you give your flag variables slightly more descriptive names you won't need such verbose comments, and the code will be more readable where they are used.

For example, if this var was instead something like `self.external_clock = False` then the usage on line 224 could be `if self.external_clock:` and the reader wouldn't need to remember which value represented which mode.

Reply:
Changed